extensions/graphql/README.md

```diff
@@ -0,0 +1,245 @@
+    this.expressMiddleware('middleware.express.GraphQL', server.expressApp);
+    this.configure(GraphQLBindings.GRAPHQL_SERVER).to({
+      asMiddlewareOnly: true,
+    });
+
+    // ...
+
+    // Customize @loopback/boot Booter Conventions here
+    this.bootOptions = {
+      graphqlResolvers: {
+        // Customize ControllerBooter Conventions here
+        dirs: ['graphql-resolvers'],
+        extensions: ['.js'],
+        nested: true,
+      },
+    };
+  }
+}
+```
+
+## Add a resolver
+
+Please add `recipe-resolver.ts` to `src/graphql-resolvers`.
+
+```ts
+import {service} from '@loopback/core';
+import {repository} from '@loopback/repository';
+import {
+  arg,
+  fieldResolver,
+  Int,
+  mutation,
+  query,
+  resolver,
+  root,
+  ResolverInterface,
+} from '@loopback/graphql';
+import {RecipeInput} from '../graphql-types/recipe-input';
+import {Recipe} from '../graphql-types/recipe-type';
+import {RecipeRepository} from '../repositories';
+import {RecipeService} from '../services/recipe.service';
+
+@resolver(of => Recipe)
+export class RecipeResolver implements ResolverInterface<Recipe> {
+  constructor(
+    // constructor injection of service
+    @repository('RecipeRepository')
+    private readonly recipeRepo: RecipeRepository,
+    @service(RecipeService) private readonly recipeService: RecipeService,
+  ) {}
+
+  @query(returns => Recipe, {nullable: true})
+  async recipe(@arg('recipeId') recipeId: string) {
+    return this.recipeRepo.getOne(recipeId);
+  }
+
+  @query(returns => [Recipe])
+  async recipes(): Promise<Recipe[]> {
+    return this.recipeRepo.getAll();
+  }
+
+  @mutation(returns => Recipe)
+  async addRecipe(@arg('recipe') recipe: RecipeInput): Promise<Recipe> {
+    return this.recipeRepo.add(recipe);
+  }
+
+  @fieldResolver()
+  async numberInCollection(@root() recipe: Recipe): Promise<number> {
+    const index = await this.recipeRepo.findIndex(recipe);
+    return index + 1;
+  }
+
+  @fieldResolver()
+  ratingsCount(
+    @root() recipe: Recipe,
+    @arg('minRate', type => Int, {defaultValue: 0.0})
+    minRate: number,
+  ): number {
+    return this.recipeService.ratingsCount(recipe, minRate);
+  }
+}
+```
+
+## Use LoopBack dependency injections in resolver classes
+
+All of LoopBack decorators for dependency injection , such as `@inject`,
+`@service`, `@repository`, and `@config`, can be used with resolver classes.
+
+```ts
+import {service} from '@loopback/core';
+
+@resolver(of => Recipe)
+export class RecipeResolver implements ResolverInterface<Recipe> {
+  constructor(
+    // constructor injection of service
+    @repository('RecipeRepository')
+    private readonly recipeRepo: RecipeRepository,
+    @service(RecipeService) private readonly recipeService: RecipeService,
+  ) {}
+}
+```
+
+## Discover and load GraphQL resolvers
+
+The `GraphQLComponent` contributes a booter that discovers and registers
+resolver classes from `src/graphql-resolvers` during `app.boot()`.
```

Comment:
Please explain what is the project layout convention - where is the booter looking for files with resolver classes.